### ISSUE TITLE
[Feature request] BatchNorm "auto" mode for use_global_states

### ISSUE THREAD

**Issue by YutingZhang** [apache/incubator-mxnet#14826]
In https://github.com/apache/incubator-mxnet/pull/14816, we have explored a gluon implementation. However, the conclusion is that an implementation directly on MxNet operator is preferable. 

It will be great if someone would like to implement this. Thanks!

**Comment by wkcn** [comment]
I think BatchNorm Op has the 'auto' mode.
https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cc#L115
